src/core/defs.jl

```diff
@@ -567,6 +577,11 @@ function variable_dimensions(obj::AbstractCompositeComponentDef, comp_path::Comp
     return dim_names(var)
 end
```

Comment:
Should you also add the same docstring to this one?

Reply:
Yea I can add it to the two other methods.  I wasn't quite clear on what a `ComponentPath` type was etc. but can write a docstring for that without a problem.

Reply:
I took a look at other doc strings for inspiration, this is done!